e2e/reservation/reservation.e2e-spec.ts

```diff
@@ -0,0 +1,360 @@
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    app.useGlobalPipes(new ValidationPipe(ValidationOptions));
+    app.useGlobalFilters(new PrismaClientExceptionFilter());
+    await app.init();
+
+    prisma = moduleFixture.get<PrismaService>(PrismaService);
+  });
+
+  beforeEach(async () => {
+    mentor = await prisma.user.create({
+      data: {
+        email: 'ReservationMentor@gmail.com',
+        nickname: 'ReservationMentor',
+        profileImage: 'ReservationMentor.png',
+        role: 'USER',
+      },
+    });
+
+    mentee = await prisma.user.create({
+      data: {
+        email: 'ReservationMentee@gmail.com',
+        nickname: 'ReservationMentee',
+        profileImage: 'ReservationMentee.png',
+        role: 'USER',
+      },
+    });
+
+    dummyMentee = await prisma.user.create({
+      data: {
+        email: 'ReservationDummyMentee@gmail.com',
+        nickname: 'ReservationDummyMentee',
+        profileImage: 'ReservationDummyMentee.png',
+        role: 'USER',
+      },
+    });
+
+    category = await prisma.category.create({
+      data: {
+        name: 'ReservationTestCategory',
+      },
+    });
+
+    hashtag = await prisma.hashtag.create({
+      data: {
+        name: 'ReservationTestHashtag',
+      },
+    });
+
+    await prisma.mentorProfile.create({
+      data: {
+        userId: mentor.id,
+        shortDescription: 'ReservationMentorShortDescription',
+        description: 'ReservationMentorDescription',
+        isHide: false,
+        mentoringCount: 0,
+      },
+    });
+
+    const mentorResponse = await request(app.getHttpServer()).get(`/dev/login/${mentor.id}`);
+    const menteeResponse = await request(app.getHttpServer()).get(`/dev/login/${mentee.id}`);
+    const dummyMenteeResponse = await request(app.getHttpServer()).get(
+      `/dev/login/${dummyMentee.id}`,
+    );
+
+    mentorAccessToken = mentorResponse.body.accessToken;
+    menteeAccessToken = menteeResponse.body.accessToken;
+    dummyMenteeAccToken = dummyMenteeResponse.body.accessToken;
+  });
+
+  afterEach(async () => {
+    await prisma.mentorFeedback.deleteMany({
+      where: {
+        OR: [{ menteeId: mentee.id }, { mentorId: mentor.id }],
+      },
+    });
+
+    await prisma.menteeFeedback.deleteMany({
+      where: {
+        OR: [{ menteeId: mentee.id }, { mentorId: mentor.id }],
+      },
+    });
+
+    await prisma.reservation.deleteMany({
+      where: {
+        mentorId: mentor.id,
+      },
+    });
+
+    await prisma.mentorProfile.deleteMany({
+      where: {
+        userId: mentor.id,
+      },
+    });
+
+    await prisma.user.deleteMany({
+      where: {
+        OR: [{ id: mentee.id }, { id: mentor.id }, { id: dummyMentee.id }],
+      },
+    });
+
+    await prisma.category.deleteMany({
+      where: {
+        id: category.id,
+      },
+    });
+
+    await prisma.hashtag.deleteMany({
+      where: {
+        id: hashtag.id,
+      },
+    });
+
+    await prisma.$disconnect();
```

Comment:
이 부분이 궁금한데, 병렬 인스턴스로 돌아서 disconnect 하게 되는건가요?

Reply:
docker 안에서 돌아가고 있는 서버의 prisma client입니다. 위 코드에서 보시면 testing모듈에서 만들어진 모듈로부터 .get 메서드 호출합니다.

```ts
    /**
     * Retrieves an instance of either injectable or controller, otherwise, throws exception.
     * @returns {TResult}
     */
    get<TInput = any, TResult = TInput>(typeOrToken: Type<TInput> | Function | string | symbol): TResult;
```

Reply:
초기화된 인스턴스 가져오는검다

Reply:
prisma.$disconnect 가 개별 테스트 종료마다 이루어지는 것이 궁금합니다.
말씀하신대로 beforeAll 에서 prisma를 가져오는 반면, afterEach에서 개별적으로 disconnect를 수행하고 있어서 궁금해졌습니다.
마찬가지로 app.close()도 궁금하네용 참고할만한 문서 있으면 추천 부탁드려요

Reply:
수정했습니다. 왜 돌아갔는지는 저도 의문이네요